drivers/serial/uart_sf32lb.c

```diff
@@ -208,10 +354,45 @@ static int uart_sf32lb_init(const struct device *dev)
 		return ret;
 	}
 
+#if CONFIG_UART_INTERRUPT_DRIVEN
+	config->irq_config_func(dev);
+#endif /* CONFIG_UART_INTERRUPT_DRIVEN */
+
 	return 0;
 }
 
+#if defined(CONFIG_UART_INTERRUPT_DRIVEN)
+static void uart_sf32lb_isr(const struct device *dev)
+{
+	struct uart_sf32lb_data *data = dev->data;
+
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+	if (data->irq_callback) {
+		data->irq_callback(dev, data->cb_data);
+	}
+#endif /* CONFIG_UART_INTERRUPT_DRIVEN */
+}
+
+#define SF32LB_UART_IRQ_HANDLER_DECL(index)                                                        \
+	static void uart_sf32lb_irq_config_func_##index(const struct device *dev);
+
+#define SF32LB_UART_IRQ_HANDLER(index)                                                             \
+	static void uart_sf32lb_irq_config_func_##index(const struct device *dev)                  \
+	{                                                                                          \
+		IRQ_CONNECT(DT_INST_IRQN(index), DT_INST_IRQ(index, priority), uart_sf32lb_isr,    \
+			    DEVICE_DT_INST_GET(index), 0);                                         \
+		irq_enable(DT_INST_IRQN(index));                                                   \
+	}
```

Comment:
this does not define any handler, just a config helper. Consider merging this into main block using `IF_ENABLED(CONFIG_XXX, (...))`.